tests/privacy_simple_network.py

```diff
@@ -0,0 +1,280 @@
+Utils.setIrreversibleTimeout(timeout)
+
+try:
+    TestHelper.printSystemInfo("BEGIN")
+    cluster.setWalletMgr(walletMgr)
+    Print("SERVER: {}".format(TestHelper.LOCAL_HOST))
+    Print("PORT: {}".format(port))
+
+    cluster.killall(allInstances=killAll)
+    cluster.cleanup()
+    Print("Stand up cluster")
+
+    if cluster.launch(pnodes=pnodes, totalNodes=totalNodes, prodCount=1, onlyBios=False, dontBootstrap=dontBootstrap, configSecurityGroup=True) is False:
+        cmdError("launcher")
+        errorExit("Failed to stand up eos cluster.")
+
+    Print("Validating system accounts after bootstrap")
+    cluster.validateAccounts(None)
+
+    def getNodes():
+        global cluster
+        nodes = [cluster.biosNode]
+        nodes.extend(cluster.getNodes())
+        return nodes
+
+#    cluster.biosNode.kill(signal.SIGTERM)
+
+    def reportInfo():
+        Utils.Print("\n\n\n\n\nInfo:")
+        for node in getNodes():
+            Utils.Print("Info: {}".format(json.dumps(node.getInfo(), indent=4, sort_keys=True)))
+
+    Utils.Print("\n\n\n\n\nCheck after KILL:")
+    Utils.Print("\n\n\n\n\nNext Round of Info:")
+    reportInfo()
+
+    producers = [cluster.getNode(x) for x in range(pnodes) ]
+    apiNodes = [cluster.getNode(x) for x in range(pnodes, totalNodes)]
+    apiNodes.append(cluster.biosNode)
+
+    def createAccount(newAcc):
+        producers[0].createInitializeAccount(newAcc, cluster.eosioAccount)
+        ignWallet = cluster.walletMgr.create("ignition")  # will actually just look up the wallet
+        cluster.walletMgr.importKey(newAcc, ignWallet)
+
+    # numAccounts = 4
+    # testAccounts = Cluster.createAccountKeys(numAccounts)
+    # accountPrefix = "testaccount"
+    # for i in range(numAccounts):
+    #     testAccount = testAccounts[i]
+    #     testAccount.name  = accountPrefix + str(i + 1)
+    #     createAccount(testAccount)
+
+    blockProducer = None
+
+    def verifyInSync(producerNum):
+        Utils.Print("Ensure all nodes are in-sync")
+        lib = producers[producerNum].getInfo()["last_irreversible_block_num"]
+        headBlockNum = producers[producerNum].getBlockNum()
+        headBlock = producers[producerNum].getBlock(headBlockNum)
+        global blockProducer
+        if blockProducer is None:
+            blockProducer = headBlock["producer"]
+        Utils.Print("headBlock: {}".format(json.dumps(headBlock, indent=4, sort_keys=True)))
+        headBlockId = headBlock["id"]
+        error = None
+        for prod in producers:
+            if prod == producers[producerNum]:
+                continue
+
+            if prod.waitForBlock(headBlockNum, reportInterval = 1) is None:
+                error = "Producer node failed to get block number {}. Current node info: {}".format(headBlockNum, json.dumps(prod.getInfo(), indent=4, sort_keys=True))
+                break
+
+            if prod.getBlock(headBlockId) is None:
```

Comment:
I see that intent here was to wait for block by id to ensure there are no forks. Maybe we can add this code to `Node.py` ? We could reuse method `waitForBlock` and if block parameter is string we consider this was block id otherwise block number. Or Alternatively we could add `waitForBlockId`

Reply:
I think that is a good idea, but would rather not tack that on here. The issue is that code relies on block numbers, so either we would have to pull in logic from nodeos to get the block num from the id, or else pass that in as well.